torchgen/gen_functionalization_type.py

```diff
@@ -84,21 +84,21 @@ def gen_composite_view_copy_kernel(g: NativeFunctionsViewGroup) -> Optional[str]
     # clone() calls in their graph (which is normally needed by reshape).
     if str(g.view_copy.func.name) == "view_copy":
         return """\
-at::Tensor view_copy(const at::Tensor & self, at::SymIntArrayRef size) {
-  // TODO: don't cast to int array ref
-  auto int_size = c10::asIntArrayRefSlow(size);
-  DimVector shape = infer_size_dv(int_size, self.numel());
+at::Tensor view_copy(const at::Tensor & self, at::IntArrayRef size) {
+  DimVector shape = infer_size_dv(size, self.numel());
   if (!at::detail::computeStride(self.sizes(), self.strides(), shape).has_value()) {
-    return self.reshape(int_size);
+    return self.reshape(size);
   } else {
-    auto output = at::_ops::view::call(self, size);
+    auto output = at::_ops::view::call(self, c10::SymIntArrayRef::fromIntArrayRef(size));
```

Comment:
ok what's going on in this kernel? the `view_copy` name above didn't get `_symint` tacked on, and some of the symint related stuff got removed, but then we are constructing symints and invoking view.